### PR TITLE
Use 'int64' instead of non-portable 'long'

### PR DESCRIPTION
**Pull request by k-dominik**
Unfortunately 'long' does not mean the same thing in different osses/compilers. On windows long will only be 32 bit (int32). Setting width explicitly is more portable.

The default dtype for in in numpy on windows is int32, but for pytorch int64... anyway. Cross-entropy loss will fail with 'long' on windows.

Bonus: Added CI to test win, macos ❤️

### CODE REVIEW

**Comment by wolny** [comment]
Thx @k-dominik, du bist klasse!

File: .github/workflows/conda-build.yml [hunk 1]
```diff
@@ -22,7 +25,7 @@ jobs:
       with:
         auto-activate-base: true
         activate-environment: ""
-        channels: local,conda-forge,defaults
+        channels: local,conda-forge
```
Comment (wolny) [inline .github/workflows/conda-build.yml:28]: looks like we need add `miniconda-version: "latest"` to this action to make it work on macos

File: .github/workflows/conda-build.yml [hunk 2]
```diff
@@ -10,8 +10,11 @@ on:
```
Comment (wolny) [inline .github/workflows/conda-build.yml:10]: looks like we need to replace `RELEASE_VERSION=$(date +%Y%m%d%H%M%S)` by something that runs on `windows-latest` runner

**Review by wolny — COMMENTED** [review]
almost there! we just need to fix the macos and windows runners :)

File: .github/workflows/conda-build.yml (hunk 2 above)
Comment (k-dominik) [reply .github/workflows/conda-build.yml:10]: good catch - also - it looks like the versioning scheme has recently been changed to calver, but no release has been done yet. I would strongly argue against date-based versioning. Meaningful pinning of this package for people that use it as a library is basically impossible with calver. Also it does not allow you to reason about versions in any meaningful way.

Of course this is also not automatically given with semver, and I have been fooled by relying on it as well. But most of the time I find it more useful than not. Only for end-user softwares (ubuntu) I can see a benefit in using calver. So... While orthogonal to this, would you be willing to reconsider?

**Comment by k-dominik** [comment]
I've added a commit reverting back to the original versioning scheme, also reading the metadata from setup.py -> no need for the environment variable shenanigans. As it stands this repo would have produced packages and code with versions out of sync (conda package calver, code would rely on __init__.py...)

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -1,6 +1,8 @@
+{% set setup_py_data = load_setup_py_data() %}
```
Comment (wolny) [inline conda-recipe/meta.yaml:1]: nice one!

File: .github/workflows/conda-build.yml (hunk 2 above)
Comment (wolny) [reply .github/workflows/conda-build.yml:10]: completely agree! It should be semver everywhere

**Review by wolny — APPROVED** [review]
great! kudos for making the versioning consistent 🚀

**Comment by k-dominik** [comment]
I can fix the windows tests later - should be easy but need to do something else.

**Comment by k-dominik** [comment]
okay weird, the numpy<2 bin should not be necessary - could you trigger the pipeline to run please @wolny

**Comment by k-dominik** [comment]
Okay, removed the numpy<2 pin as I could at least track down the environment solving problem to the `nvidia` channel. I am pretty sure I had an issue open somewhere in the pytorch org but it seems that the respective repo got privated... Not sure exactly which library is shadowing which, but strict solving only works _without_ the nvidia channel.

**Comment by k-dominik** [comment]
this is good to go imo.

**Comment by wolny** [comment]
amazing that you've made the build work on all 3 OSes! Thanks again for this PR 🚀
